### DISCUSSION TITLE
advantage of udp over esp now

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#1605] [Q&A] [answered]
Thanks for the beautiful library . 

After working with espnow for month i realize there was distance and line of sight issues (i change antennas / boards etc etc ) but still i get stable distance of few meters .  so that make my mind to work with UDP  

after go through this video of atomic14 . he says UDP is fast and long range too due to router  so by this we can say UDP is better if we have router ?  

https://www.youtube.com/watch?v=d_h38X4_eQQ

in this video at ( 0:04:00 time )  atomic prove UDP is better   

but after some basic home work still i am confuse should i go with UDP or not ? i need fast and long range communication  .  or is their any better option ??

your opinion and point of view is very import for all  of us .

**Comment by pschatzmann** [comment, marked as answer]
I don't know: I did not do any distance tests.
